File: Lecture07/ViewPager/app/src/main/java/com/codingblocks/viewpager/MainActivity.java
package com.codingblocks.viewpager;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ViewPager viewPager = findViewById(R.id.viewPager);

        ArrayList<Fragment> fragments = new ArrayList<>();
        fragments.add(new FragmentA());
        fragments.add(new FragmentB());
        fragments.add(new FragmentC());

        PagerAdapter pagerAdapter = new PagerAdapter(getSupportFragmentManager(), fragments);
        viewPager.setAdapter(pagerAdapter);

        TabLayout tabLayout = findViewById(R.id.tabLayout);

        tabLayout.setupWithViewPager(viewPager, true);

    }
}

File: Lecture12/PokemonApp/app/src/main/java/com/codingblocks/pokemonapp/MainActivity.kt
package com.codingblocks.pokemonapp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.R.string
import android.util.Log
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.*
import java.io.IOException


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val client = OkHttpClient()

        val request = Request.Builder()
            .url("https://pokeapi.co/api/v2/pokemon/1")
            .build()

        client.newCall(request).enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) = Unit

            override fun onResponse(call: Call, response: Response) {

                val responseString: String? = response.body()?.string()

                responseString?.let { resp ->
                    Log.e("TAG", resp)
                    //Parsing the responseString using GSON
                    //Setting the arraylist to the adapter
                    //Notifying the adapter

                    val gson = Gson()
                    val pokemonResponse = gson.fromJson(resp, PokemonResponse::class.java)

                    Log.e("TAG", pokemonResponse.sprites.backDefault)

                    runOnUiThread {
                        tView.text = pokemonResponse.name.capitalize()
                    }

                } ?: kotlin.run {
                    Log.e("TAG", "Request failed")
                }
//                if (responseString != null) {
//                    //Parsing the responseString using GSON
//                    //Setting the arraylist to the adapter
//                    //Notifying the adapter
//                } else {
//                    //Show an error message
//                }
            }

        })

    }


}
